client/post-editor/media-modal/secondary-actions.jsx

```diff
@@ -80,7 +82,11 @@ export default React.createClass( {
 			} );
 		}
 
-		if ( ModalViews.GALLERY !== activeView && selectedItems.length && userCan( 'upload_files', site ) ) {
+		const canDeleteItems = selectedItems.length && ! selectedItems.some( ( item ) => {
```

Comment:
The negation here was a bit tricky for me to read, so maybe an extra variable could help:
```javascript
const userCanDeleteAllItems = ! selectedItems.some( ( item ) => {
       return ! canUserDeleteItem( item, user, site );
} );
const showDeleteButton = selectedItems.length && userCanDeleteAllItems;